### ISSUE TITLE
Cannot conditionally deliver feature based on OpenGL version

### ISSUE THREAD

**Issue by benjaminRomano** [google/bundletool#130]
**Describe the bug**
We generate a dynamic feature module with the following condition:
```
                    <dist:device-feature	
                        dist:name="android.hardware.opengles.version"	
                        dist:version="0x00030001"	
                        />
```

but when trying to extract an APK set for a device with opengl version greater than or equal to that value it does not included it in the output APK set. It appears it tries to do a comparison of `android.hardware.opengles.version: 0x00030001` against `reqGlEsVersion: 0x30002` [here](https://github.com/google/bundletool/blob/master/src/main/java/com/android/tools/build/bundletool/device/DeviceFeatureMatcher.java#L47). 

Would this also be a problem in Play store builds as well?

**Bundletool version(s) affected**
bundletool 0.9.0 and extracted with 0.10.2. 

**To Reproduce**
1. Create app bundle with dynamic feature with opengl device feature condition
2. Build APK set
3. extract apk set for device-spec that should have proper OpenGL version . I tested with a device spec generated from a Pixel 3.
4. Observe output directory does not included the dynamic feature's split apks.

**Expected behavior**
Bundletool extract-apk logic properly handles the openGL case. This would include both of the following scenarios:
1. openGL version is equivalent
2. device openGL version is greater than that specified in device feature version.

**Known workaround**
None.

**Environment:**
Mac

**Comment by keyboardsurfer** [comment]
I could not verify this behavior with bundletool 0.11 and the public sample for [Dynamic Features](https://github.com/android/app-bundle-samples/blob/master/DynamicFeatures). The [AndroidManifest](https://github.com/android/app-bundle-samples/blob/master/DynamicFeatures/features/native/src/main/AndroidManifest.xml) was adapted to include the device-feature changes you outlined above and set to `dist:onDemand="false"`.
With a Pixel 3 spec I got these apk extracted from the apks:

```
base-en.apk
base-master.apk
base-xxhdpi.apk
initialInstall-master.apk
initialInstall-xxhdpi.apk
native-arm64_v8a.apk
native-master.apk
native-xxhdpi.apk
split-master.apk
split-xxhdpi.apk
url-master.apk
url-xxhdpi.apk
```

**Comment by plecesne** [comment]
Benjamin, can you double-check that you effectively had `dist:onDemand="false"` set?
If you're still facing issues, please reopen the bug.

**Comment by benjaminRomano** [comment]
I updated our toolchain to use bundletool 12 and this now works as expected.

**Comment by benjaminRomano** [comment]
@plecesne @keyboardsurfer 
I updated the native feature module's `AndroidManifest.xml` with what we were using for our own feature module internally. So the native feature module's `AndroidManifest.xml` now looks like this:
```
    <dist:module dist:title="@string/module_native">
        <dist:delivery>
            <dist:install-time>
                <dist:conditions>
                        <dist:device-feature	
                            dist:name="android.hardware.opengles.version"	
                            dist:version="0x00030001"	
                        />

                    <dist:min-sdk dist:value="24" />

                </dist:conditions>
            </dist:install-time>
        </dist:delivery>
        <dist:fusing dist:include="false" />
    </dist:module>

```
The expectation being that this module is included at install-time for devices that are SDK 24+ and have a minimum openGL version of 3.1, but still have the potential to be dynamically delivered.

Next, I built and extracted the apk sets for my device (Pixel 3). The built apk set has the native module as expected. However, the `extract-apks` does not include the `native` module.

<img width="1673" alt="Screen Shot 2019-12-31 at 1 15 56 PM" src="https://user-images.githubusercontent.com/9221137/71634355-26a8a580-2bd0-11ea-8300-cb7bf0e152e8.png">

My assumption is that the `extract-apks` outputs the set of apks included at install time, but maybe that's not correct?

**Comment by keyboardsurfer** [comment]
I created a [branch for this on the samples](https://github.com/android/app-bundle-samples/tree/bundletool130).

Then using

```
./gradlew bundle
cd app/build/outputs/bundle/debug
bundletool build-apks --bundle app.aab --output app.apks
bundletool extract-apks --apks app.apks --device-spec ~/.android/spec/pixel3.json
```

Creates the files I mentioned in [above comment](https://github.com/google/bundletool/issues/130#issuecomment-568101458)

Can you please verify the behavior on your end with the [official sample branch](https://github.com/android/app-bundle-samples/tree/bundletool130)?

**Comment by benjaminRomano** [comment]
Yep I get the same output as you on that branch. However, I want my module to be conditionally delivered *and* have the possibility to be dynamically delivered. In your example, `dist:on-demand="false"` is set, which I wouldn't want.

**Comment by keyboardsurfer** [comment]
I can verify that when `dist:onDemand="true"` bundletool does not create the apks with above instructions.

**Comment by keyboardsurfer** [comment]
When using an `on-demand` module and the `--modules` flag for `extract-apks` I get the module to be extracted from the bundle.

**Comment by benjaminRomano** [comment]
Will this affect play store's delivery of modules with `openGL` requirements?

**Comment by keyboardsurfer** [comment]
This will likely affect all modules with conditional and ondemand delivery. I recommend marking it only as conditional. If it's not downloaded due to the device condition but you feel it should have you can still download it later on using the PlayCore API.

**Comment by ymakhno** [comment]
Fixed in 1.5.0.
